.devcontainer/devcontainer.json

```diff
@@ -48,5 +59,5 @@
     // Mount the parent as /workspaces so we can pip install peers as editable
     "workspaceMount": "source=${localWorkspaceFolder}/..,target=/workspaces,type=bind",
     // After the container is created, install the python project in editable form
-    "postCreateCommand": "pip install $([ -f dev-requirements.txt ] && echo '-c dev-requirements.txt') -e '.[dev]' && pre-commit install"
+    "postCreateCommand": "uv sync --frozen && uvx pre-commit install && SHELL=bash uv tool update-shell"
```

Comment:
Some questions:
- Why `--frozen`? I would argue it is safer to do it without, as that will potentially update your lockfile if you added a dependency to `pyproject.toml` which you should at least know about before deciding whether it was intentional or not and committing the result
- Why `uvx` rather than `uv run`? Pre-commit should be listed in the dev dependencies, and this will ensure we get a consistent version of it.
- Do we use any uv tools? Why did you need to run this command to add the tool dir to the path?

`uv sync` I get, `--frozen` I think makes sense, but why `uvx pre-commit install` rather than `uv run pre-commit install`?
Also what does

Reply:
Not sure I follow the frozen comment. `--frozen` doesn't update the lockfile and ensures the environment you have is the same as the one expected. Without it I couldn't get it to not need `--prerelease=allow` to deal with dodal et al depending on alpha versions of things and I didn't want that to be something we encourage.

I can't remember the motivation for `uvx` rather than `uv run`, might have been related to making pre-commit available everywhere without setting the `$PATH` to the venv as you did for ophyd-async. I will try it without and see what fails.

> Also what does

I'm assuming there was meant to be more to this question